File: golang/3074.go
package main

import "sort"

func minimumBoxes(apple []int, capacity []int) int {
    boxes := len(capacity) + 1

    apples := 0
    for _, apple := range apple {
        apples += apple
    }

    sort.Ints(capacity)

    capSum := 0
    for _, c := range capacity {
        capSum += c
    }

    count := 0
    for _, c := range capacity {
        if capSum >= apples {
            capSum -= c
            count++
        }
    }

    return boxes - count
}

func main () {
	println(minimumBoxes([]int{1, 3, 2}, []int{4,3,1,5,2}))
}

File: js/3074.js
/**
 * @param {number[]} apple
 * @param {number[]} capacity
 * @return {number}
 */
const minimumBoxes = function(apple, capacity) {
    const totalApples = apple.reduce((acc, count) => acc+count, 0)
    const boxes = capacity.map(x => x).sort((a, b) => b-a)

    let boxRequired = 0
    let applesLeft = totalApples
    
    for (let i = 0; i < boxes.length; i++) {
        const cap = boxes[i]
        applesLeft -= cap
        boxRequired++
        if (applesLeft <= 0) {
            break
        }
    }

    return boxRequired
};

console.log(minimumBoxes([5,5,5], [2,4,2,7])) // 4